### PR TITLE
[16.0] [MIG] product_tax_multicompany_default

### PR DESCRIPTION
**Pull request by Shide**
Straightforward migration

Things changed:
- Groups id of view on field
- Performing an SQL to remove all taxes from other companies (cannot do it with sudo or SUPERUSER_ID) (done in a separate Commit to be able to cherry-pick)

MT-1564 @moduon @rafaelbn @carlosdauden @cubells

### CODE REVIEW

**Review by yajo — APPROVED** [review]
Code OK!

**Review by loida-vm — APPROVED** [review]
If in most of the invoices the tax is X, this is the one that will be suggested in the product, when we are going to create a new one.
If in a company in the database, it does not have any invoice yet, when we are going to create the first product, it does not suggest any tax, because it does not recognize which one is the most used. 

The button in the product view to set taxes in other companies, only appears if there is more than one company in the DB.

LGTM! 👌

**Comment by rafaelbn** [comment]
Hello @carlosdauden , do you agree fixing this BUG?

https://www.loom.com/share/0f50c6ee2dd04c0f80390a2bd94362e9

@Shide we should FIX this BUG in 15 ASAP please 🙏 

Thank you!

MT-2121 @moduon

**Comment by carlosdauden** [comment]
:+1:

File: product_tax_multicompany_default/models/product.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        excl_supplier_tax_ids: List[int] = None,
+    ):
+        """Delete taxes from product excluding chosen taxes
+
+        :param excl_customer_tax_ids: Excluded customer tax ids
+        :param excl_supplier_tax_ids: Excluded supplier tax ids
+        """
+        tax_where = " AND tax_id NOT IN %s"
+        # Delete customer taxes
+        customer_sql = "DELETE FROM product_taxes_rel WHERE prod_id IN %s"
```
Comment (carlosdauden) [inline product_tax_multicompany_default/models/product.py:42]: Why use SQL?
Set customer_tax_ids  and supplier_tax_ids as you like before than this code:
```
        self.write(
            {
                "taxes_id": [(6, 0, customer_tax_ids)],
                "supplier_taxes_id": [(6, 0, supplier_tax_ids)],
            }
        )

```

**Review by carlosdauden — CHANGES_REQUESTED** [review]

File: product_tax_multicompany_default/models/product.py (hunk 1 above)
Comment (Shide) [reply product_tax_multicompany_default/models/product.py:42]: Cannot do it with sudo or SUPERUSER_ID, don't know why, but I've tried a lot of possibilities with sudo, with_user(SUPERUSER_ID), with_context(allowed_company_ids=...), with_company(), and nothing...

Systematically is excluing all the [(5, None)] on other companies. 
This is the reason why I've chosen to empty vía sql

**Review by loida-vm — APPROVED** [review]
Functional review :+1:

**Review by fcvalgar — APPROVED** [review]
LGTM 
Thank you @Shide

**Review by carlosdauden — APPROVED** [review]
I'm not very convinced, but I don't want to block :wink:

**Pull request merged**